From 7f79da2edb12aea277ea9e6498f73ecf7a5d671e Mon Sep 17 00:00:00 2001
From: Alejandro Abdelnur <tucu@apache.org>
Date: Wed, 16 Jan 2013 23:56:04 +0000
Subject: [PATCH 007/596] MR1: MAPREDUCE-4315. jobhistory.jsp throws 500 when
 a .txt file is found in /done. (sandyr via tucu)

git-svn-id: https://svn.apache.org/repos/asf/hadoop/common/branches/branch-1@1434506 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit 61ee40b37429c41d9cbc4c619b6db4a12644dfa0)
(cherry picked from commit a6ccb440ce302fcfc6dc64ad167eba08b60da6d4)
(cherry picked from commit a035acecc775f6718b1253cd90b66e8bf89ac945)
(cherry picked from commit 9867370c9e8ff9e2e2b65897ee181220d9966d71)
(cherry picked from commit 34154c6a66c85c9a7d9d35b850cd87074815d475)
---
 .../src/webapps/job/jobhistory.jsp                 |   16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/hadoop-mapreduce1-project/src/webapps/job/jobhistory.jsp b/hadoop-mapreduce1-project/src/webapps/job/jobhistory.jsp
index c237941..558045b 100644
--- a/hadoop-mapreduce1-project/src/webapps/job/jobhistory.jsp
+++ b/hadoop-mapreduce1-project/src/webapps/job/jobhistory.jsp
@@ -124,9 +124,23 @@ window.location.href = url;
         //  - jobname contains the keyword
         return "".equals(uqJobname) || fileName.split("_")[FILENAME_JOBNAME_PART].toLowerCase().contains(uqJobname);
       }
+      
+      private boolean isHistoryFile(String fileName) {      	
+        String[] tokens = null;
+        try {
+          String dp = JobHistory.JobInfo.decodeJobHistoryFileName(fileName);
+          tokens = dp.split("_");
+        } catch (IOException ioe) {
+        }
+
+        return tokens != null && !fileName.endsWith(".xml") && tokens.length > 3
+            && tokens[1].matches("\\d+")  && tokens[2].matches("\\d+")
+            && tokens[3].matches("\\d+");
+      }
+
 
       public boolean accept(Path path) {
-        return !(path.getName().endsWith(".xml")) && matchUser(path.getName()) && matchJobName(path.getName());
+        return isHistoryFile(path.getName()) && matchUser(path.getName()) && matchJobName(path.getName());
       }
     };
     
-- 
1.7.9.5

